product_contract/models/sale_order.py

```diff
@@ -109,6 +104,14 @@ def action_create_contract(self):
                 order_lines.write({"contract_id": contract.id})
             for line in line_to_update_contract:
                 line.create_contract_line(line.contract_id)
+            if lines_without_contract:
+                contract = contract_model.create(
+                    rec._prepare_contract_value(self.env["contract.template"])
+                )
+                contracts.append(contract.id)
+                contract._onchange_contract_type()
```

Comment:
Something to do on contract side to put a TODO to change `automatic_price` on lines to computed.

Reply:
Isn't the same? When you put `env[model]`, a new recordset is instantiated, so no possible pool sharing between successive calls to this.